api-reference/10 UI Widgets/dxDiagram/1 Configuration/nodes/customDataExpr.md

```diff
@@ -5,10 +5,14 @@ default: undefined
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+Specifies the name of a data source field or an expression that provides a node's custom data.
 
 ##### param(data): Object
-<!-- Description goes here -->
+The current node's data object.
 
 ---
-<!-- Description goes here -->
\ No newline at end of file
+This option links custom data contained in a data source to a diagram node. The linked data is copied and pasted with the node. Changes in the data are reflected in a diagram history and can be undone and redone by the UI.
```

Comment:
"The linked data is copied and pasted with the node."

Can you clarify what you mean here?

Reply:
the same as above

Reply:
```suggestion
This option links custom data from a data source to a diagram node. The node contains the linked data copied from the data source. Changes in the data are reflected in the diagram history. You can use the UI to undo and redo these changes.
```